samcli/cli/cli_config_file.py

```diff
@@ -0,0 +1,152 @@
+"""
+CLI configuration decorator to use TOML configuration files for click commands.
+"""
+
+## This section contains code copied and modified from [click_config_file][https://github.com/phha/click_config_file/blob/master/click_config_file.py]
+## SPDX-License-Identifier: MIT
+
+import functools
+import os
+import logging
+
+import click
+import toml
+
+__all__ = ("TomlProvider", "configuration_option")
+
+LOG = logging.getLogger(__name__)
+DEFAULT_CONFIG_FILE_NAME = "sam-config"
+DEFAULT_IDENTIFER = "default"
+
+
+class TomlProvider:
+    """
+    A parser for toml configuration files
+    :param cmd: sam command name as defined by click
+    :param section: section defined in the configuration file nested within `cmd`
+    """
+
+    def __init__(self, cmd=None, section=None):
+        self.cmd = cmd
+        self.section = section
+
+    def __call__(self, file_path, identifier, cmd_name):
+        """
+        :param file_path: The path to the configuration file
+        :param identifier: The name of the sectional identifier within configuration file.
+        :param cmd_name: sam command name as defined by click
+        :returns dictionary containing the configuration parameters under specified identifier
+        """
+        config = toml.load(file_path)
```

Comment:
- What happens if the file does not exist?
- What happens if the TOML file cannot be parsed?

Reply:
there are checks for that.